libsolidity/ExpressionCompiler.cpp

```diff
@@ -800,6 +800,47 @@ void ExpressionCompiler::LValue::storeValue(Expression const& _expression, bool
 	}
 }
 
+void ExpressionCompiler::LValue::setToZero(Expression const& _expression) const
+{
+	switch (m_type)
+	{
+	case STACK:
+	{
+		unsigned stackDiff = m_context->baseToCurrentStackOffset(unsigned(m_baseStackOffset));
+		if (stackDiff > 16)
+			BOOST_THROW_EXCEPTION(CompilerError() << errinfo_sourceLocation(_expression.getLocation())
+												  << errinfo_comment("Stack too deep."));
+		else if (stackDiff > 0)
+			for (unsigned i = 0; i < m_size; ++i)
+				*m_context << u256(0) << eth::Instruction::POP;
```

Comment:
I think there is a SWAP missing here.
Please also add a test that detects this problem, i.e. just use delete on a local variable.